gef-ui/server/controllers/application-details/index.test.js

```diff
@@ -343,40 +371,148 @@ describe('controllers/application-details', () => {
 
         await applicationDetails(mockRequest, mockResponse);
 
-        expect(mockResponse.render)
-          .toHaveBeenCalledWith('partials/application-preview.njk', expect.objectContaining({
+        expect(mockResponse.render).toHaveBeenCalledWith(
+          'partials/application-preview.njk',
+          expect.objectContaining({
             applicationStatus: mockApplicationResponse.status,
-          }));
+          }),
+        );
       });
 
       it('renders `review-decision` when page requested is `review-decision`', async () => {
         mockApplicationResponse.status = CONSTANTS.DEAL_STATUS.UKEF_APPROVED_WITHOUT_CONDITIONS;
         api.getApplication.mockResolvedValueOnce(mockApplicationResponse);
-
-        await applicationDetails(MOCKS.MockRequestUrl('/gef/application/123/review-decision'), mockResponse);
-
-        expect(mockResponse.render)
-          .toHaveBeenCalledWith('partials/review-decision.njk', expect.objectContaining({
-            applicationStatus: mockApplicationResponse.status,
-          }));
+        const req = {
+          ...MOCKS.MockRequestUrl('/gef/application/123/review-decision'),
+          flash: jest.fn().mockReturnValue('Facility is updated'),
+        };
+        await applicationDetails(req, mockResponse);
+
+        expect(mockResponse.render).toHaveBeenCalledWith(
+          'partials/review-decision.njk',
+          expect.objectContaining({ applicationStatus: mockApplicationResponse.status }),
+        );
+        expect(req.flash).toHaveBeenCalledWith('success');
       });
 
       it('renders `unissued-facilities` when page requested is `unissued facilities` ', async () => {
         mockApplicationResponse.status = CONSTANTS.DEAL_STATUS.UKEF_ACKNOWLEDGED;
         api.getApplication.mockResolvedValueOnce(mockApplicationResponse);
+        const req = {
+          ...MOCKS.MockRequestUrl('/gef/application/123/unissued-facilities'),
+          flash: jest.fn().mockReturnValue('Facility is updated'),
```

Comment:
We shouldn't need this flash mock here as we're not testing for the behavior when req.flash is present in this test? Ie. it should work without this.

Reply:
If removed test fails after removing flash

Reply:
once we have fixed our mocking of flash -- we should be able to use our helper function to mock successful flash response here (and everywhere where needed in this file)